### PR TITLE
Cu 32q93wp add a btn on fbt pdp page to add every product at once 

### PR DESCRIPTION
**Pull request by VirAlex**
## Objective
What: Add all items in one, into the cart

## Type
- [x] New Feature

screencapture-localhost-5173-search-product-EV421I0BV-2022-10-18-18_49_22
screencapture-localhost-5173-search-product-EV421I0BV-2022-10-18-18_49_12

### CODE REVIEW

**Comment by diegolarraz** [comment]
Task linked: [CU-32q93wp Add a btn on FBT (PDP page) to add every product at once](https://app.clickup.com/t/32q93wp)

**Comment by matthew1809** [comment]
Made some small fixes @VirAlex, especially on the add all button

**Comment by matthew1809** [comment]
Do you have object IDs to test where there is one, two, three and more FBT products? Like to test the edge cases around this

**Comment by VirAlex** [comment]
> Do you have object IDs to test where there is one, two, three and more FBT products? Like to test the edge cases around this

In fact like it's not a Carousel, we limited to 2 FBT results + 1 original hit so it will never have more than 3 cards to display

**Comment by matthew1809** [comment]
> > Do you have object IDs to test where there is one, two, three and more FBT products? Like to test the edge cases around this
> 
> In fact like it's not a Carousel, we limited to 2 FBT results + 1 original hit so it will never have more than 3 cards to display

Perfect, I'm happy with this PR, but would prefer at least one other person test and approve as well, because it's a big piece of functionality, maybe @HugoWith .

Great job, @VirAlex!!!

File: src/pages/productDetailsPage/ProductDetails.jsx [hunk 1]
```diff
@@ -104,6 +106,8 @@ const ProductDetails = () => {
     }
   }, []);
 
+  useEffect(() => {}, [hit]);
```
Comment (diegolarraz) [inline src/pages/productDetailsPage/ProductDetails.jsx:109]: probably we dont need this, as the useEffect is empty its redundant, changes on `hit` should force a re-render even without this.

**Review by diegolarraz — APPROVED** [review]
Have a quick look at the comments, otherwise it looks very good!

File: src/pages/productDetailsPage/ProductDetails.jsx [hunk 2]
```diff
@@ -325,14 +332,22 @@ const ProductDetails = () => {
                 />
               </div>
             )}
-          {shouldHaveFbtProductsValue && fbtRecommendationsProducts.length > 0 && (
-            <div>
+          {shouldHaveFbtProductsValue && fbtRecommendationsProducts.length > 1 && (
+            <>
```
[the remark this answers is not in the document]
Comment (matthew1809) [reply src/pages/productDetailsPage/ProductDetails.jsx:336]: Because we add the current PDP product into to that array, and we don't want to show FBT if there's just the current product but no recommendations, there will always be one, even if there are no recommendations
